tests/lib/ecma-version.js

```diff
@@ -196,6 +196,50 @@ describe("ecmaVersion", () => {
             }, /'char' is reserved/u);
         });
 
+        it("Should throw error when using ES3, allowReserved: false, and reserved words", () => {
+            assert.throws(() => {
+                espree.parse(
+                    "var char = 'c'", {
+                        ecmaVersion: 3,
+                        allowReserved: false
+                    }
+                );
+            }, /'char' is reserved/u);
+        });
+
+        it("Should not throw error when using ES3, allowReserved: true, and reserved words", () => {
+            assert.doesNotThrow(() => {
+                espree.parse(
+                    "var char = 'c'", {
+                        ecmaVersion: 3,
+                        allowReserved: true
+                    }
+                );
+            }, /'char' is reserved/u);
+        });
+
+        it("Should not throw error when using ES3, allowReserved: true, and reserved words in object literals", () => {
+            assert.doesNotThrow(() => {
+                espree.parse(
+                    "var x = { char: 'c' }", {
+                        ecmaVersion: 3,
+                        allowReserved: true
+                    }
+                );
+            }, /'char' is reserved/u);
+        });
+
+        it("Should not throw error when using ES5, allowReserved: false, and ES3 reserved words in object literals", () => {
+            assert.doesNotThrow(() => {
+                espree.parse(
+                    "var x = { char: 'c' }", {
+                        ecmaVersion: 5,
+                        allowReserved: false
+                    }
+                );
+            }, /'char' is reserved/u);
+        });
+
```

Comment:
good to have a test: `{ecmaVersion: 5, allowReserved: true}` to assert it throws.

Reply:
Done.